src/utils/__tests__/generateLimitValues.unit.test.ts

```diff
@@ -0,0 +1,47 @@
+import {
+  generateLimitValues,
+} from '..';
+import {
+  InternalValidationError,
+} from '../../errors';
+
+describe('generateLimitValues', () => {
+  it('should return calculated values when valid parameters are provided', () => {
+    const limitValues = generateLimitValues({
+      count: 100,
+      page: 42,
+    });
+    expect(limitValues).toEqual({
+      limit: 100,
+      offset: 4100,
+    });
+  });
+
+  it('should error if count is a float value', () => {
+    expect(() => generateLimitValues({
+      count: 100.5,
+      page: 42,
+    })).toThrow(InternalValidationError);
+  });
+
+  it('should error if count is less than 1', () => {
+    expect(() => generateLimitValues({
+      count: 0,
+      page: 42,
+    })).toThrow(InternalValidationError);
+  });
+
+  it('should error if page is less than 1', () => {
+    expect(() => generateLimitValues({
+      count: 100.5,
+      page: 0,
+    })).toThrow(InternalValidationError);
```

Comment:
This would fail validation for multiple reasons, wouldn't it?

```suggestion
  it('should error if page is less than 1', () => {
    expect(() => generateLimitValues({
      count: 100,
      page: 0,
    })).toThrow(InternalValidationError);
```

Reply:
Good catch -- my initial implementation was zero indexed and I didn't update this test.